cluster-autoscaler/cloudprovider/gce/gce_url.go

```diff
@@ -64,24 +61,36 @@ func ParseInstanceUrlRef(url string) (GceRef, error) {
 }
 
 // GenerateInstanceUrl generates url for instance.
-func GenerateInstanceUrl(ref GceRef) string {
+func GenerateInstanceUrl(domainSuffix string, ref GceRef) string {
+	if domainSuffix == "" {
+		domainSuffix = defaultGceDomainSuffix
+	}
+	instanceUrlTemplate := prefix(domainSuffix) + "%s/zones/%s/instances/%s"
 	return fmt.Sprintf(instanceUrlTemplate, ref.Project, ref.Zone, ref.Name)
 }
 
 // GenerateMigUrl generates url for instance.
-func GenerateMigUrl(ref GceRef) string {
+func GenerateMigUrl(domainSuffix string, ref GceRef) string {
+	if domainSuffix == "" {
+		domainSuffix = defaultGceDomainSuffix
+	}
+	migUrlTemplate := prefix(domainSuffix) + "%s/zones/%s/instanceGroups/%s"
 	return fmt.Sprintf(migUrlTemplate, ref.Project, ref.Zone, ref.Name)
 }
 
+func prefix(domainSuffix string) string {
```

Comment:
Can we allow to override the whole API endpoint? It allows slightly more flexibility while shortening the code slightly.

Also, I believe "API endpoint" wording is more obvious than "domain suffix" when reading the code :)

Reply:
changed it to domain url